pkg/openstack/ansibleee.go

```diff
@@ -0,0 +1,65 @@
+package openstack
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
+
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	ansibleeev1 "github.com/openstack-k8s-operators/openstack-ansibleee-operator/api/v1alpha1"
+	corev1beta1 "github.com/openstack-k8s-operators/openstack-operator/apis/core/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+// ReconcileAnsibleEE -
+func ReconcileAnsibleEE(ctx context.Context, instance *corev1beta1.OpenStackControlPlane, helper *helper.Helper) (ctrl.Result, error) {
+	if !instance.Spec.AnsibleEE.Enabled {
+		return ctrl.Result{}, nil
+	}
+
+	ansibleEE := &ansibleeev1.OpenStackAnsibleEE{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "ansibleee", //FIXME
+			Namespace: instance.Namespace,
+		},
+	}
+
+	helper.GetLogger().Info("Reconciling AnsibleEE", "AnsibleEE.Namespace", instance.Namespace, "ansibleEE.Name", "ansibleee")
+	op, err := controllerutil.CreateOrPatch(ctx, helper.GetClient(), ansibleEE, func() error {
+		err := controllerutil.SetControllerReference(helper.GetBeforeObject(), ansibleEE, helper.GetScheme())
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+
+	if err != nil {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			corev1beta1.OpenStackControlPlaneAnsibleEEReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			corev1beta1.OpenStackControlPlaneAnsibleEEReadyErrorMessage,
+			err.Error()))
+		return ctrl.Result{}, err
+	}
+	if op != controllerutil.OperationResultNone {
+		helper.GetLogger().Info(fmt.Sprintf("AnsibleEE %s - %s", ansibleEE.Name, op))
+	}
+
+	// TODO: if ansibleEE.IsReady() {
```

Comment:
@jlarriba @slagle
the other operators implement `IsReady^, should we implement it on Ansible EE?

Reply:
@fao89 I'm not sure about the purpose of `IsReady` in this context: in general that function is used to check the Status of the pods deployed by the operator and `Mark` the `Status` with a given message.
With `AnsibleEE`, you're building and running `Jobs`: they can be tracked during the execution with an `UID`, but at the end the job will be automatically set to `Completed`.
Not saying we don't need that kind of function, just not sure about the purpose.

Reply:
I agree with @fmount